services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/NeoBolt.java

```diff
@@ -0,0 +1,210 @@
+
+            String status = record.get("state").asString();
+            SwitchState st = "active".equals(status) ? SwitchState.ACTIVATED : SwitchState.DEACTIVATED;
+            sw.setState(st);
+
+            results.add(sw);
+        }
+        LOGGER.info("Result: {}", results.size());
+
+        return results;
+    }
+
+    private List<IslInfoData> getISLs(Session session) {
+        LOGGER.debug("Getting ISLs...");
+        String q =
+                "MATCH (:switch)-[isl:isl]->(:switch) " +
+                        "RETURN " +
+                        "isl.src_switch as src_switch, " +
+                        "isl.src_port as src_port, " +
+                        "isl.dst_switch as dst_switch, " +
+                        "isl.dst_port as dst_port, " +
+                        "isl.speed as speed, " +
+                        "isl.max_bandwidth as max_bandwidth, " +
+                        "isl.latency as latency, " +
+                        "isl.available_bandwidth as available_bandwidth, " +
+                        "isl.status as status " +
+                        "order by isl.src_switch";
+
+        StatementResult queryResults = session.run(q);
+        List<IslInfoData> results = new ArrayList<>();
+        for (Record record : queryResults.list()) {
+            LOGGER.debug("Reading record... {}", record);
+            // max_bandwidth not used in IslInfoData
+            List<PathNode> pathNodes = new ArrayList<>();
+            PathNode src = new PathNode();
+            src.setSwitchId(record.get("src_switch").asString());
+            src.setPortNo(parseValue(record.get("src_port")));
+            src.setSegLatency(parseValue(record.get("latency")));
+            pathNodes.add(src);
+            PathNode dst = new PathNode();
+            dst.setSwitchId(record.get("dst_switch").asString());
+            dst.setPortNo(parseValue(record.get("dst_port")));
+            dst.setSegLatency(parseValue(record.get("latency")));
+            pathNodes.add(dst);
+
+            String status = record.get("status").asString();
+            IslChangeType state = ("active".equals(status)) ? IslChangeType.DISCOVERED : IslChangeType.FAILED;
+
+            IslInfoData isl = new IslInfoData(
+                    parseValue(record.get("latency")),
+                    pathNodes,
+                    parseValue(record.get("speed")),
+                    state,
+                    parseValue(record.get("available_bandwidth"))
+            );
+            isl.setTimestamp(System.currentTimeMillis());
+
+            results.add(isl);
+            LOGGER.info("Result ISL: {}", isl);
+
+        }
+        return results;
+    }
+
+
+    @Override
+    public void declareOutputFields(OutputFieldsDeclarer declarer) {
+        declarer.declare(new Fields("response", "correlationId"));
+    }
+
+    @Override
+    public void cleanup() {
+        driver.close();
+    }
+
+    private int parseValue(Value value) {
```

Comment:
Should it be named as "parseIntValue"?
PS Looks like it can't handle numbers stored as strings... it is pity but we have them in DB.